core/src/main/java/nl/nn/adapterframework/doc/DocWriterNewXmlUtils.java

```diff
@@ -214,6 +215,26 @@ static XmlBuilder addAttribute(XmlBuilder context, String name, String typeName)
 		return attribute;
 	}
 
+	static XmlBuilder addAttribute(XmlBuilder context, String name, AttributeType modelAttributeType) {
+		XmlBuilder attribute = new XmlBuilder("attribute", "xs", XML_SCHEMA_URI);
+		attribute.addAttribute("name", name);
+		String typeName = "xs:";
+		switch(modelAttributeType) {
+		case BOOL:
+			typeName += "boolean";
+			break;
+		case INT:
+			typeName += "integer";
+			break;
+		case STRING:
+			typeName += "string";
+			break;
+		}
```

Comment:
Door op deze manier het type te zetten (en iets dergelijks geldt ook voor enums), wordt het onmogelijk om integer en enum waarden te configureren als bijvoorbeeld ``portNum="${portNum}"``. 
Is daar een workaround voor?

Reply:
Opgelost voor integer en boolean, nog niet voor enum

Reply:
Commit gepushed voor enums, maar die werkt niet. Ik ga een nieuwe commit maken om het te fixen.

Reply:
Nu zou die ook voor enums goed moeten zijn. Ik heb de integratie testen gedraaid door even de assume... uit te commentariëren. Ik heb ook even uit een gegenereerde XSD geknipt en geplakt. Ik heb de constructie met xs:union getest met de online XSD validator. Hij staat inderdaad zowel enum waardes als variabele referenties toe.